src/server/svr_chk_owner.c

```diff
@@ -390,66 +390,94 @@ authenticate_user(struct batch_request *preq, struct connection *pcred)
  *		this checks the most conditions common to most job batch requests.
  *		It also returns a pointer to the job if found and the tests pass.
  *		If the request is for a single subjob or a range of subjobs (of an
- *		Job Array),  the return job pointer is to the parent Array Job.
+ *		Job Array), alsoe return pointer to the parent Array Job.
  * @par
  *		Depending on what the "jobid" identifies, the following is returned
- *		in the integer pointed to by rc:
+ *		in the integer pointed to by jobtype:
  *	   	IS_ARRAY_NO (0)       - for a regular job
  *	   	IS_ARRAY_ArrayJob (1) - for an Array Job
  *	   	IS_ARRAY_Single (2)   - for a single subjob
  *	   	IS_ARRAY_Range (3)    - for a range of  subjobs
  *
  * @param[in]	jobid	-	Job Id.
  * @param[in,out]	preq	-	job batch request
- * @param[out]	rc	-	Depending on what the "jobid" identifies,
+ * @param[out]	jobtype	-	Depending on what the "jobid" identifies,
  * 						the following is returned
  *						(0)       - for a regular job
  *						(1) - for an Array Job
  *						(2)   - for a single subjob
  *						(3)    - for a range of  subjobs
+ * @param[out]	jobstate	-	the state of the job
+ * @param[out]	ppjob	-	pointer to the job
+ * @param[out]	parent	-	pointer to the parent job of subjob
+ * @param[out]	poffset	-	offset of subjob in tracking table
  *
- * @return	job *
- * @retval	a pointer to the job	: if found and the tests pass.
- * @retval	NULL	: failed
+ * @return	int
+ * @retval	TRUE	: if found and the tests pass.
+ * @retval	FALSE	: test failed
  */
 
 
-job *
-chk_job_request(char *jobid, struct batch_request *preq, int *rc)
+int
+chk_job_request(char *jobid, struct batch_request *preq, int *jobtype, int *jobstate, job **ppjob, job **parent, int *poffset)
```

Comment:
Please don't change the prototype unless there is a strong reason to do so. Please look into my detailed reply for similar comment above.

Reply:
As explained I have consolidated and collated similar code from various req_xxx() functions to make chk_job_request() more justifiable to its goal while decoupling all its returns, we can safely add new parameters to help speed up and optimize things. Further I feel the assumption made is a bit unwarranted to have any considerations here. Let me know if you still feel the parameters are unnecessary and there is something to justify the assumptions made on the future changes.

Reply:
There seemed to be no performance difference by making the change to this function, neither any need for any refactoring on this area. The existing function is simpler and the code at the callers are simple as well. There is a bit of duplicate code in the various req_xxx handlers dealing with array jobs, but that can be broken out to functions later, and is not a concern of this particular enhancement. In the interest for lower risk, the code should be reverted back to its original form.

Reply:
I understand. Reverted in review comit (rc 20):  https://github.com/PBSPro/pbspro/pull/590/commits/b01fcac3e94363966f635e6fad726a5342288a79

Reply:
I also had to revert check_deletehistoryjob() due to this

Reply:
Thanks - this definitely reduces risk on the project